spotify_tensorflow/luigi/python_dataflow_task.py

```diff
@@ -0,0 +1,301 @@
+        self._output_uris = {}
+
+        for (name, target) in self._output.items():
+            uri = target.generate_uri() if hasattr(target, "generate_uri") else get_uri(target)
+            uri = uri.rstrip("/")
+            output_args.append("--%s=%s" % (name, uri))
+            self._output_uris[name] = uri
+
+        return output_args
+
+    def _get_runner(self):
+        return "DirectRunner" if self.local_runner else "DataflowRunner"
+
+    def _get_dataflow_args(self):
+        dataflow_args = []
+
+        _runner = self._get_runner()
+        if _runner:
+            dataflow_args += ["--runner={}".format(_runner)]
+        if self.project:
+            dataflow_args += ["--project={}".format(self.project)]
+        if self.staging_location:
+            dataflow_args += ["--staging_location={}".format(self.staging_location)]
+        if self.zone:
+            dataflow_args += ["--zone={}".format(self.zone)]
+        if self.region:
+            dataflow_args += ["--region={}".format(self.region)]
+        if self.temp_location:
+            dataflow_args += ["--temp_location={}".format(self.temp_location)]
+        if self.num_workers:
+            dataflow_args += ["--num_workers={}".format(self.num_workers)]
+        if self.autoscaling_algorithm:
+            dataflow_args += ["--autoscaling_algorithm={}".format(self.autoscaling_algorithm)]
+        if self.max_num_workers:
+            dataflow_args += ["--max_num_workers={}".format(self.max_num_workers)]
+        if self.network:
+            dataflow_args += ["--network={}".format(self.network)]
+        if self.subnetwork:
+            dataflow_args += ["--subnetwork={}".format(self.subnetwork)]
+        if self.disk_size_gb:
+            dataflow_args += ["--disk_size_gb={}".format(self.disk_size_gb)]
+        if self.worker_machine_type:
+            dataflow_args += ["--worker_machine_type={}".format(self.worker_machine_type)]
+        if self.job_name:
+            dataflow_args += ["--job_name={}".format(self.job_name)]
+        if self.worker_disk_type:
+            dataflow_args += ["--worker_disk_type={}".format(self.worker_disk_type)]
+        if self.service_account:
+            dataflow_args += ["--service_account_email={}".format(self.service_account)]
+        if self.requirements_file:
+            dataflow_args += ["--requirements_file={}".format(self.requirements_file)]
+
+        return dataflow_args
+
+    def args(self):
+        """ Extra arguments that will be passed to your dataflow job.
+
+        Example:
+            return ["--project=my-gcp-project",
+                    "--zone=a-zone",
+                    "--staging_location=gs://my-gcp-project/dataflow"]
+
+        Note that:
+
+            * You "set" args by overriding this method in your subclass.
+            * This function should return an iterable of strings.
+        """
+        return []
+
+    def get_output_uris(self):
+        return self._output_uris
```

Comment:
What's this function used by? Should it be moved up and given a doctoring (given it's "public")?

Reply:
Mostly it will be in user luigi task. I want to expose `_output_uris` because in some cases user may want to dump data to different subdirectory under the same output uri, eg. training/evaluation

Reply:
I'll add a description